pkg/cmd/auth/gitcredential/helper.go

```diff
@@ -0,0 +1,117 @@
+package login
+
+import (
+	"bufio"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type config interface {
+	Get(string, string) (string, error)
+}
+
+type CredentialOptions struct {
+	IO     *iostreams.IOStreams
+	Config func() (config, error)
+
+	Operation string
+}
+
+func NewCmdCredential(f *cmdutil.Factory, runF func(*CredentialOptions) error) *cobra.Command {
+	opts := &CredentialOptions{
+		IO: f.IOStreams,
+		Config: func() (config, error) {
+			return f.Config()
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:    "git-credential",
+		Args:   cobra.ExactArgs(1),
+		Short:  "Implements git credential helper protocol",
```

Comment:
Are we worried at all about this protocol updating and us having to maintain/upkeep this implementation?

Reply:
The oldest reference to this specific format of the git credential protocol that I could find is from 2012, and it hasn't changed since. It's also very unlikely that git will ever change it—at least not in a backwards-incompatible way—because it's in git's strong interest to avoid breaking current programs that have implemented the protocol.